python/lsst/ts/m2/csc.py

```diff
@@ -239,18 +243,23 @@ async def handle_summary_state(self):
 
         # Run the mock server in the simulation mode
         # self.simulation_mode is the attribute from upstream: BaseCsc
-        if self.simulation_mode == 1 and self._mock_server is None:
+        if (self.summary_state == salobj.State.STANDBY) and (
+            self._mock_server is not None
+        ):
+            await self._mock_server.close()
+            self._mock_server = None
+
+        if (self.simulation_mode == 1) and (self._mock_server is None):
```

Comment:
Yes. This is the behavior I am looking for. The `MockServer` relies on the`tcpip.OneClientServer`. I realized it can support the multiple times connections but the `callback` may not in my test. Therefore, this is a workaround here. Thanks!